addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/connector/WebInterface.java

```diff
@@ -0,0 +1,276 @@
+ */
+package org.openhab.binding.solaredge.internal.connector;
+
+import static org.openhab.binding.solaredge.SolarEdgeBindingConstants.*;
+
+import java.io.UnsupportedEncodingException;
+import java.net.CookieStore;
+import java.net.HttpCookie;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpStatus.Code;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.solaredge.config.SolarEdgeConfiguration;
+import org.openhab.binding.solaredge.handler.SolarEdgeHandler;
+import org.openhab.binding.solaredge.internal.command.PostLoginGetClientCookie;
+import org.openhab.binding.solaredge.internal.command.PostLoginGetSpringSecurityToken;
+import org.openhab.binding.solaredge.internal.command.PreLogin;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The connector is responsible for communication with the solaredge webportal
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+public class WebInterface {
+
+    private static final long LOGIN_WAIT_TIME = 1000;
+
+    private final Logger logger = LoggerFactory.getLogger(WebInterface.class);
+
+    /**
+     * Configuration of the bridge from
+     * {@link org.openhab.BoxHandler.fritzaha.handler.FritzAhaBridgeHandler}
+     */
+    private final SolarEdgeConfiguration config;
+
+    /**
+     * Bridge thing handler for updating thing status
+     */
+    private final SolarEdgeHandler handler;
+
+    /**
+     * holds authentication status
+     */
+    private boolean authenticated = false;
+
+    /**
+     * holds cookie status
+     */
+    private boolean cookieOK = false;
+
+    /**
+     * HTTP client for asynchronous calls
+     */
+    private HttpClient asyncclient;
+    /**
+     * Maximum number of simultaneous asynchronous connections
+     */
+    private int asyncmaxconns = 20;
+
+    /**
+     * Constructor to set up interface
+     *
+     * @param config Bridge configuration
+     */
+    public WebInterface(SolarEdgeConfiguration config, SolarEdgeHandler handler) {
+        this.config = config;
+        this.handler = handler;
+        asyncclient = new HttpClient(new SslContextFactory(true));
+        asyncclient.setMaxConnectionsPerDestination(asyncmaxconns);
+        try {
+            asyncclient.start();
+        } catch (Exception e) {
+            logger.error("Could not start HTTP Client");
+        }
+        authenticate();
+    }
+
+    /**
+     * executes any command provided by parameter
+     *
+     * @param command
+     */
+    public void executeCommand(SolarEdgeCommand command) {
+        if (!isAuthenticated()) {
+            authenticate();
+            try {
+                Thread.sleep(LOGIN_WAIT_TIME);
+            } catch (InterruptedException e) {
+                assert true;
```

Comment:
What is the intention here?

Reply:
I just want to do nothing, I cannot remember why I used "assert true" but as far as I remember it won't compile to any bytecode.